charts/metering-ansible-operator/templates/crds/meteringconfig.crd.yaml

```diff
@@ -1549,6 +1549,29 @@ spec:
                                   properties:
                                     useInstanceCredentials:
                                       type: boolean
+                            prometheus:
+                              type: object
+                              properties:
+                                enabled:
+                                  type: boolean
+                                uri:
+                                  description: URI for Prometheus.
+                                  type: string
+                                chunkSizeDuration:
+                                  description: Default size of each Prometheus query chunk.
+                                  type: string
+                                maxQueryRangeDuration:
+                                  description: Default range for Prometheus query, divided into chunkSizeDuration chunks.
+                                  type: string
+                                cacheDuration:
+                                  description: Time to live for Presto checking connector properties values.
+                                  type: string
+                                bearer:
+                                  type: object
+                                  properties:
+                                    tokenFile:
+                                      description: Path to token file for bearer authentication to Prometheus.
+                                      type: string
```

Comment:
We may be able to get away with just validating that the `uri` property is set when `enabled: true`, and not set when `enabled: false`, e.g.:

```yaml
                              oneOf:
                              - required:
                                - uri
                                properties:
                                  enabled:
                                    enum:
                                    - true
                              - allOf:
                                - not:
                                    required:
                                    - uri
                                properties:
                                  enabled:
                                    enum:
                                    - false
```

Reply:
In 42681d7 we add another subschema to match enabled: false with there not being a uri defined. We also define format uri for uri.